apps/meteor/app/slackbridge/server/SlackAdapter.js

```diff
@@ -1006,13 +1006,12 @@ export default class SlackAdapter {
 				],
 			};
 
-			if (!isImporting) {
-				await Messages.setPinnedByIdAndUserId(
-					`slack-${slackMessage.attachments[0].channel_id}-${slackMessage.attachments[0].ts.replace(/\./g, '-')}`,
-					rocketMsgObj.u,
-					true,
-					new Date(parseInt(slackMessage.ts.split('.')[0]) * 1000),
-				);
+			if (!isImporting && slackMessage.attachments[0].channel_id && slackMessage.attachments[0].ts) {
+				const messageId = `slack-${slackMessage.attachments[0].channel_id}-${slackMessage.attachments[0].ts.replace(/\./g, '-')}`;
+				await Messages.setPinnedByIdAndUserId(messageId, rocketMsgObj.u, true, new Date(parseInt(slackMessage.ts.split('.')[0]) * 1000));
+				if (settings.get('Message_Read_Receipt_Store_Users')) {
```

Comment:
A suggestion, maybe having the setting check inside the service and decide there if doing something is better than checking for the setting on each place we want to use it? 👀

Reply:
Since in this case I'm calling the raw model directly you mean adding the setting check in the raw model code? :thinking: I haven't seen us doing that in other places

Reply:
No no, thought this was a call to the ReadReceipts service 😬 

If it's the model, there's nothing much we can do, unless we wrap the calls in the service. Not sure if that would be in scope for this one tho

Reply:
Since the `MessageReads` service is only used for read receipts in threads as of now, I guess we can keep this out of scope for now. But it'd certainly be a great improvement once we include all of the read-receipts features in the `MessageReads` service (or any other)